src/routes/api.php

```diff
@@ -32,10 +32,22 @@
         Route::delete('presentations/{presentation}/users/{user}', 'API\PresentationUserController@unsubscribe')->name('presentations.users.unsubscribe');
         // manage user rights on presentations
         Route::put('presentations/{presentation}/users/{user}', 'API\PresentationUserController@changeRole')->name('presentations.change_role');
+
         // poll management
         Route::apiResource('presentations.polls', 'API\PollController')->shallow();
-        Route::put('polls/{poll}/publish', 'API\PollController@publish')->name('polls.publish');
-        Route::get('polls/{poll}/results', 'API\PollController@results')->name('polls.results');
+        Route::middleware(['poll.role:' . App\User\Role::PRESENTER()])->group(function () {
+            Route::put('presentations/{presentation}/polls', 'API\PollController@store')->name('presentations.polls.store');
+            Route::put('polls/{poll}', 'API\PollController@update')->name('polls.update');
+            Route::put('polls/{poll}/publish', 'API\PollController@publish')->name('polls.publish');
+            Route::get('polls/{poll}/results', 'API\PollController@results')->name('polls.results');
+            Route::post('polls/{poll}/choices', 'API\ChoiceController@store')->name('polls.choices.store');
+        });
+        Route::middleware(['poll.role'])->group(function () {
+            Route::get('presentations/{presentation}/polls', 'API\PollController@index')->name('presentations.polls.index');
+            Route::get('polls/{poll}', 'API\PollController@show')->name('polls.show');
+            Route::get('polls/{poll}/choices', 'API\ChoiceController@index')->name('polls.choices.index');
```

Comment:
that route is already generated by line 52

Reply:
Fixed in aceb9a4